### PR TITLE
refactor(pubsub): add no-op addmessage to batch sink interface

### PR DESCRIPTION
**Pull request by alevenberg**
Part of https://github.com/googleapis/google-cloud-cpp/issues/13171

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13174)

### CODE REVIEW

File: google/cloud/pubsub/internal/default_batch_sink.h [hunk 1]
```diff
@@ -39,6 +39,9 @@ class DefaultBatchSink : public BatchSink {
 
   ~DefaultBatchSink() override = default;
 
+  /// No-op implementation.
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/default_batch_sink.h:42]: I'd remove this comment. I can see it is a noop on the next line.

File: google/cloud/pubsub/internal/sequential_batch_sink.h [hunk 2]
```diff
@@ -40,6 +40,9 @@ class SequentialBatchSink
 
   ~SequentialBatchSink() override = default;
 
+  /// No-op implementation.
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/sequential_batch_sink.h:43]: same

File: google/cloud/pubsub/internal/batch_sink.h [hunk 3]
```diff
@@ -38,6 +39,9 @@ class BatchSink {
  public:
   virtual ~BatchSink() = default;
 
+  /// Adds the message to the batch the message.
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/batch_sink.h:42]: s/batch the message./batch./

File: google/cloud/pubsub/internal/batch_sink.h [hunk 4]
```diff
@@ -38,6 +39,9 @@ class BatchSink {
...
+  virtual void AddMessage(pubsub::Message m) = 0;
```
Comment (dbolduc) [inline google/cloud/pubsub/internal/batch_sink.h:43]: This is the new `SaveMessage()`, right?

https://github.com/googleapis/google-cloud-cpp/blob/945644596dfc9a7a6bcc3dac484f9c2a1a7b76f5/google/cloud/pubsub/internal/batching_publisher_connection.cc#L57

If we aren't going to `std::move()` the message in, we should be using a `pubsub::Message const& m`. That saves a copy in the no-op cases.

**Comment by alevenberg** [comment]
Resolved addressed comments

**Review by dbolduc — APPROVED** [review]

**Pull request merged**